src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1033,6 +1034,45 @@ public void importDataset(final RelationKey relationKey, final Schema schema, fi
   }
 
   /**
+   * @param relationKey the relationalKey of the dataset to import
+   * @return the status
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+
+    /* delete from postgres at each worker by calling the DbDelete operator */
+    try {
+      Map<Integer, SubQueryPlan> workerPlans = new HashMap<>();
+      for (Integer workerId : getWorkersForRelation(relationKey, null)) {
+        workerPlans.put(workerId, new SubQueryPlan(new DbDelete(EmptyRelation.of(catalog.getSchema(relationKey)),
+            relationKey, null)));
+      }
+      ListenableFuture<Query> qf =
+          queryManager.submitQuery("delete " + relationKey.toString(), "delete " + relationKey.toString(),
```

Comment:
So I'm concerned about the transactional integrity of this distributed delete. I know we don't really do distributed transactions in Myria, but I don't really see a way around it here. I know the Postgres JDBC driver supports "XA" (two-phase commit), but I've never used it, and I'm not sure how it would map to our subquery/operator model. I know that `DROP TABLE` is transactional in Postgres, so at least you don't have to do `TRUNCATE TABLE` and `DROP TABLE` in separate steps.